src/network/networkd-link.c

```diff
@@ -2176,6 +2179,15 @@ static int link_update_name(Link *link, sd_netlink_message *message) {
         if (streq(ifname, link->ifname))
                 return 0;
 
+        if (!format_ifname(link->ifindex, ifname_from_index))
+                return log_link_debug_errno(link, SYNTHETIC_ERRNO(ENXIO), "Could not get interface name for index %i.", link->ifindex);
+
+        if (!streq(ifname, ifname_from_index)) {
+                log_link_debug(link, "Received new interface name '%s' does not match with the current interface name '%s', ignoring.",
```

Comment:
I find this wording a bit unclear, maybe "New interface name '%s' received from the kernel does not correspond with the name currently configured on the actual interface, "%s". Ignoring."